File: 01-basics/problem-001/Solution.py
number=int(input("enter day:\n"))

while number < 1 or number > 366:
    number = int(input("Invalid day, enter again:\n"))


# The first 6 months of the Persian calendar have 31 days each.   
if number<=6*31:
    month=number//31
    if number%31!=0:
        month+=1
    print(month)
    
# Months 7 through 11 have 30 days each.
elif number>6*31 and number<=6*31 +150:
    month=6+ (number - 6 * 31) // 30
    if (number-6*31)%30 !=0:
        month+=1
    print(month)

# Any remaining valid day belongs to month 12.
else:
    print(12)
    

File: 01-basics/problem-001/Solution.java

import java.util.Scanner;

public class Solution{
    public static void main(String[] args) {
        Scanner input=new Scanner(System.in);
        
        System.out.println("enter day:\n");
        int number=input.nextInt();
        int month=0;

        // Check if the entered day is valid
        // Persian year can have up to 366 days
        while(number>366 || number<1){
            System.err.println("Invalid day, enter again:\n");
            number=input.nextInt();
        }

        // First 6 months of the Persian calendar have 31 days
        if(number<=6*31){
            month=number/31;

            // If the day is not exactly at the end of a month,
            // move to the next month
            if(number%31!=0)
                month++;
        }

        // Months 7 to 11 have 30 days
        else if(number>6*31 && number<=6*31+150){
            month=6+ (number - 6*31)/30;
            if ((number - 6*31)%30 !=0)
                month++;
        }

        // Remaining days belong to month 12 (Esfand)
        else{
            month=12;
        }
        System.out.println(month);

    }
}